prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2164,18 +2178,26 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
                     vir::Expr::not(cond_var.into())
                 };
 
+                let msg_desc =
+                    // rustc_middle/src/mir/mod.rs says BoundsCheck is expected to be handled by
+                    // the caller (?!)
```

Comment:
I guess this is because bounds check are encoded kind of specially, with all the bounds checks for a (potentially nested) projection before the actual projection happens? Does [`fmt_assert_args`](https://github.com/rust-lang/rust/blob/62652865b6029b4776a7c03efa13a37b15c9b953/compiler/rustc_middle/src/mir/mod.rs#L1343) help us here? If not, `"BoundsCheck"` is not good enough as a description for the user, since this message is what Prusti displays in the error when the verifier detects a reachable assertion failure.